ops/ruby/devops-framework/DEVOPS_FRAMEWORK.md

```diff
@@ -0,0 +1,78 @@
+# DevOps Framework
+
+## Purpose
+This is a small but flexible framework for automating DevOps actions and ensuring they are
+taken in a traceable, repeatable, and consistent way. It operates by  communicating to Google Cloud Platform
+directly, independently of the existing build, operations, and deployment pipeline(s).
+
+## Usage
+The entry point to all functionality is `devops.rb`, which provides command line and environment
+loading and dispatches to Task classes (not scripts) that do the real work.
+`./devops.rb  -t <task_name> -e <env_file> --task-option-1 value1`
+
+There is also a dynamically-generated help file from the descriptions given in the options.
+```
+$ ruby ./devops.rb  --help
+Usage: devops [options]
+    -t, --task [TASK]                Task to be in in each environment
```

Comment:
I'd have a slight preference for a positional arg rather than a flag for a task/command name, especially if there will be alternate flags for subcommands. This is pretty typical for CLIs with subcommands.